### PR TITLE
Make logfile default location .unison rather than $HOME

### PR DESCRIPTION
**Pull request by pgossman**
Addresses #318

Accessing the unison directory was not initially possible from ubase/trace.ml—where
the logfile preference is defined—because unisonDir was defined in Os. Currently,
to access the unison directory from Util, Util defines a function reference where top-level
modules supply Os.fileInUnisonDir to it (via Util.supplyFileInUnisonDirFn).

Instead of moving the logfile preference from trace.ml to some module above ubase,
this PR moves the definition of Os.unisonDir and Os.fileInUnisonDir to Util. This has the
added benefit of being able to delete three existing duplicated calls to
Util.supplyFileInUnisonDirFn. The unisonDir and fileInUnisonDir values now reference
those in Util.

### CODE REVIEW

**Review by gdt — APPROVED** [review]
Thank you for keeping the code cleanup and the functional change in separate commits.

I would like to have review from someone who knows ocaml better to ensure that the code cleanup results in the same behavior under all circumstances.    (I don't mean to imply that I think it doesn't.)

I think having the log file in .unison is a good change; I have long noticed it appearing in $HOME where it obviously doesn't belong.

I'll wait for others to comment, and will merge this anyway after 7 days even if not.   Thanks for contributing to unison.

**Review by tleedjarv — APPROVED** [review]
Looks ok, with a few comments.

I don't particularly like the code duplication with `isMacOSXPred` but I see this is difficult to avoid here.

Why have you chosen to not replace all occurrences of `Os.unisonDir` and `Os.fileInUnisonDir` with the new `Util.` functions? Either way, I suggest you move code comments from os.ml to util.ml.

**Comment by pgossman** [comment]
> Why have you chosen to not replace all occurrences of Os.unisonDir and Os.fileInUnisonDir with the new Util. functions? Either way, I suggest you move code comments from os.ml to util.ml.

I haven't worked on open-source before and I wasn't sure what is appropriate for the scope of change in a PR, so I stayed minimal. I agree that it is better to remove the values in Os and replace occurrences with the new ones in Util. 

The commit that was just pushed removes the values from Os and moves the comments as you suggested.

Thank you for reviewing.

**Comment by tleedjarv** [comment]
> I haven't worked on open-source before and I wasn't sure what is appropriate for the scope of change in a PR, so I stayed minimal.

Reasonable approach. And that's why the review is open. Everyone can help and voice their opinion if this is too much or too little. I am not a guardian of this project either.

The change itself looks good to me.

The merge commit in the PR may not be a good idea, though. I don't know how it will look like once the PR itself is merged but I would prefer your commits rebased. @gdt or anyone else, any clue or preference?

**Comment by pgossman** [comment]
> The merge commit in the PR may not be a good idea, though.

Sorry about that. I just rebased and reapplied the commits.

**Review by tleedjarv — APPROVED** [review]
All looks good to me.

**Comment by gdt** [comment]
Answering question from earlier, I have a strong preference for rebasing and a fully-cleaned up set of commits, to the point where I'd be very uncomfortable approving a merge that isn't clean.  It's not that hard to do, and it makes looking at the history far easier.  I see the rebase has happened - thanks! - but since I was asked wanted to answer.

**Review by gdt — APPROVED** [review]
This looks ok to me too.

Scope of a branch merge is always tricky, but I see all the cleanup/moving of the helper functions as clearly belonging together.   If anything, it's the move of the logfile that should be separate, but for this project, a separate commit is good enough, in the grand balance between making progress easier and making understanding what happened later easier.

**Pull request merged**
